kernel/src/main/scala/cats/kernel/instances/SortedSetInstances.scala

```diff
@@ -23,12 +23,24 @@ private[instances] trait SortedSetInstances1 {
 class SortedSetOrder[A: Order] extends Order[SortedSet[A]] {
   def compare(a1: SortedSet[A], a2: SortedSet[A]): Int =
     cats.kernel.instances.int.catsKernelStdOrderForInt.compare(a1.size, a2.size) match {
-      case 0 => StaticMethods.iteratorCompare(a1.iterator, a2.iterator)
+      case 0 =>
+        // In the event that the ordering instances are not _exactly_ the
+        // same, we have to sort/rebuild the sets. If we don't we risk
+        // violating the Monoid identity law.
+        if (a1.ordering == a2.ordering) {
```

Comment:
don't you need to compare to `Order[A]` as well? they could be the same, but not the same as `Order[A]` right?

Reply:
I don't _think_ so we _need_ to check that, but I could be wrong. This gets really tricky here, but I _think_ the laws all hold as long as the ordering used in both sets is consistent with each other, even if they are inconsistent with `Order` used to actually compare the elements. We only need _consistent_ iteration order for traversing the sets here, using our `Order` for comparing the elements.

We can't meaningfully compare `Order[A]` to the `Ordering[A]` values even if we wanted to, at least not without some additional machinery. 

FWIW, I don't really like this. As I discussed in #4103, I think we should consider adding a newtype which always and only uses the coherent `Order` for the underlying `SortedSet`, but I'm leaving that for a different PR.

Reply:
@johnynek I'm wrong. Unless all three order/orderings agree we can break transitivity. 

https://github.com/typelevel/cats/runs/4694302475?check_suite_focus=true#step:15:598

I'll take a look here.

Reply:
Well if we only need the same iteration order we could just pick the ordering of the larger set and resort only the other one.